build.gradle.kts

```diff
@@ -55,64 +57,68 @@ repositories {
 
 // region Workarounds for https://github.com/gradle/gradle/issues/22335
 tasks.register("apiDump") {
-    it.dependsOn(gradle.includedBuild("plugin").task(":apiDump"))
+    dependsOn(gradle.includedBuild("plugin").task(":apiDump"))
 }
 
 tasks.register("apiCheck") {
-    it.dependsOn(gradle.includedBuild("plugin").task(":apiCheck"))
+    dependsOn(gradle.includedBuild("plugin").task(":apiCheck"))
 }
 
 afterEvaluate {
     gradle.includedBuilds.forEach { included ->
         project(":kotlinx-benchmark-runtime").tasks.named("publishToMavenLocal") { dependsOn(included.task(":publishToMavenLocal")) }
     }
 }
+tasks.register("publishToMavenLocal") {
```

Comment:
I'm not sure, the build config is not easy to decipher.

The code in the `afterEvaluate {}` block makes the `:kotlinx-benchmark-runtime:publishToMavenLocal` task depend on the `:publishToMavenLocal` task of every included build.

This means if any other task depends on `:kotlinx-benchmark-runtime:publishToMavenLocal`, then it will also run the `:publishToMavenLocal` task of every included build. So maybe if you run `./gradlew fooTask` and that requires `:kotlinx-benchmark-runtime:publishToMavenLocal`, and that will trigger the other `:publishToMavenLocal` tasks.

The `tasks.register("publishToMavenLocal)` in the root build doesn't set up the same task dependencies, so they're not the same. It's more useful for being able to quickly run publishing to be able to check it works.

I want to come back and fix the build config, so it's easier to understand and work with. Converting the build scripts to use the Kotlin DSL is just one step of that.